modules/portfolio/lib/portfolio-data.service.ts

```diff
@@ -92,35 +92,50 @@ export class PortfolioDataService {
         const blockNumber = parseInt(block.number);
         const poolShares = [];
 
-        for (const sharesOwned of user.sharesOwned ?? []) {
-            const pool = pools.find((pool) => pool.id === sharesOwned.poolId.id);
-
-            if (!pool) {
-                continue;
+        const poolIds = [];
+        for (const farm of farmUsers) {
+            if (farm.pool) {
+                poolIds.push(farm.pool?.pair);
```

Comment:
you don't need the `?` here since you check the presence in your if statement above

Reply:
Right, removed it.